searchlib/src/vespa/searchlib/attribute/postinglistsearchcontext.hpp

```diff
@@ -68,31 +68,38 @@ PostingListSearchContextT<DataT>::fillArray()
 
 template <typename DataT>
 struct PostingListSearchContextT<DataT>::FillPart : public vespalib::Runnable {
-    FillPart(const PostingStore& posting_store, const DictionaryConstIterator & from, size_t count,
-             BitVector & bv, uint32_t limit)
-        : _posting_store(posting_store), _bv(bv),
+    FillPart(const PostingStore& posting_store, const DictionaryConstIterator & from, size_t count, uint32_t limit)
+        : FillPart(posting_store, from, count, nullptr, limit)
+    { }
+    FillPart(const PostingStore& posting_store, const DictionaryConstIterator & from, size_t count, BitVector * bv, uint32_t limit)
+        : _posting_store(posting_store),
+          _bv(bv),
           _docIdLimit(limit),
           _from(from),
           _to(from)
     {
         _to += count;
     }
     void run() override {
+        if (_bv == nullptr) {
+            _owned_bv = BitVector::create(_docIdLimit);
+            _bv = _owned_bv.get();
+        }
         for (;_from != _to;++_from) {
             addToBitVector(PostingListTraverser<PostingStore>(_posting_store, _from.getData().load_acquire()));
         }
     }
-    void addToBitVector(const PostingListTraverser<PostingStore> & postingList)
-    {
+    void addToBitVector(const PostingListTraverser<PostingStore> & postingList) {
         postingList.foreach_key([this](uint32_t key) {
-            if (__builtin_expect(key < _docIdLimit, true)) { _bv.setBit(key); }
+            if (__builtin_expect(key < _docIdLimit, true)) { _bv->setBit(key); }
         });
     }
-    const PostingStore      &_posting_store;
-    BitVector               &_bv;
-    uint32_t                 _docIdLimit;
-    DictionaryConstIterator  _from;
-    DictionaryConstIterator  _to;
+    const PostingStore        &_posting_store;
+    BitVector                 *_bv;
+    uint32_t                   _docIdLimit;
+    DictionaryConstIterator    _from;
+    DictionaryConstIterator    _to;
+    std::unique_ptr<BitVector> _owned_bv;
```

Comment:
it is polite to initialize this in the constructor as well

Reply:
Agree, fixed